src/Schema/MySQLSchemaManager.php

```diff
@@ -567,7 +572,8 @@ protected function fetchTableOptionsByTable(string $databaseName, ?string $table
            ccsa.CHARACTER_SET_NAME
       FROM information_schema.TABLES t
         INNER JOIN information_schema.COLLATION_CHARACTER_SET_APPLICABILITY ccsa
-            ON ccsa.COLLATION_NAME = t.TABLE_COLLATION
+	     ON /*M!101001 ccsa.FULL_COLLATION_NAME = t.TABLE_COLLATION OR */
+               ccsa.COLLATION_NAME = t.TABLE_COLLATION
```

Comment:
I'd like to avoid version conditionals in SQL. We have our platform class hierarchy for this purpose.

I'll accept this change as a hotfix for MariaDB 11.4 on the 3.8.x branch. Can you work on a PR for 3.9.x that moves the SQL statement to `AbstractMySQLPlatform` and that introduces a new platform class for MariaDB 10.10 which overrides the generated SQL?